content/livesync/models/models.textile

```diff
@@ -0,0 +1,291 @@
+
+The following is an example of a sync function. It retrieves the latest model state from your database:
+
+```[javascript]
+async function sync(id: number, page: number) {
+  const result = await fetch(`/api/post/${id}?page=${page}`);
+  return result.json();
+}
+```
+
+The following is an example response from the sync function:
+
+```[json]
+{
+  "sequenceID": "1",
+  "data": {
+    "id": 123,
+    "text": "Hello World",
+    "comments": []
+  }
+}
+```
+
+The sync function needs to be registered when a model is "instantiated":#create, for example:
+
+```[javascript]
+const modelsClient = modelsClient.models.get({
+  name: 'myPost',
+  channelName: 'post:123',
+  sync,
+  merge,
+});
+```
+
+Explicitly call the sync function directly on the model in order to bootstrap its state on initial page load, for example:
+
+```[javascript]
+await model.sync(123, 1);
+```
+
+If calling the sync function throws an error, the Models SDK will automatically retry according to the strategy specified by the @syncOptions.retryStrategy@ defined in your "@ClientOptions@":/livesync/models#options.
+
+h3(#sequence). SequenceID
+
+The @sequenceID@ enables the Models SDK to know the point in the stream of change events that corresponds to this version of the database state.
+
+Internally, the Models SDK will replay change events from the point specified by the @sequenceID@ onwards in order to bring the model state up to date with any realtime changes that occur from the time at which the state was read from the database.
+
+Your backend endpoint called by your sync function should return the largest @sequenceID@ in the "outbox":/livesync/connector/database#outbox at the time that the state is read:
+
+```[sql]
+-- Start a database transaction
+BEGIN;
+-- Query your model state
+SELECT * FROM …;
+-- Obtain the sequence ID
+SELECT COALESCE(MAX(sequence_id), 0) FROM outbox;
+-- Commit the database transaction
+COMMIT;
+```
+
+For this to work correctly, use a "Read Committed":https://www.postgresql.org/docs/7.2/xact-read-committed.html transaction isolation level when reading your database state. This is the default isolation level in PostgreSQL.
+
+h3(#history). Replay from history
+
+After the Models SDK initializes the model state via the sync function, it is important to apply the exact sequence of change events from your backend from the correct point in the stream. This position is specified by the @sequenceID@ returned from your backend endpoint.
+
+Internally, after calling the sync function, the SDK attaches to the "channel":/channels and "subscribes":#subscribe to live messages. Live messages are buffered and not yet processed. It then paginates backwards through the message history on the channel from the point of attachment using "@untilAttach@":/storage-history/history#until-attach. The number of items in each page can be configured via the @syncOptions.historyPageSize@ defined in your "@ClientOptions@":/livesync/models#options.
```

Comment:
The `subscribe` link here should be to _channel_ subscribe docs, not model subscribe docs

Reply:
Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)